lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -876,32 +892,53 @@ slotDifference epl a b
     a' = flatSlot epl a
     b' = flatSlot epl b
 
+-- | Return the slot immediately before the given slot.
+slotPred :: SlotParameters -> SlotId -> SlotId
+slotPred (SlotParameters (EpochLength el) _ _) (SlotId en sn)
+    | sn > 0    = SlotId (en    ) (               sn - 1)
+    | otherwise = SlotId (en - 1) (fromIntegral $ el - 1)
+
+-- | Return the slot immediately after the given slot.
+slotSucc :: SlotParameters -> SlotId -> SlotId
+slotSucc (SlotParameters (EpochLength el) _ _) (SlotId en sn)
+    | sn < el - 1 = SlotId (en    ) (sn + 1)
+    | otherwise   = SlotId (en + 1) (     0)
```

Comment:
What's the need for these two functions :thinking: ? They are unused and weren't needed before. 
Truth is, I'd rather not have to maintain code that we don't use and make the codebase bigger than necessary.

Reply:
> What's the need for these two functions thinking ? They are unused and weren't needed before.

Their existence makes it easier to express the following two laws:
```hs
slotSucc . slotStartingAtOrJustBefore . utcTimePred . slotStartTime == id
slotPred . slotStartingAtOrJustAfter  . utcTimeSucc . slotStartTime == id
```

Without `slotPred` and `slotSucc`, we could use `slotDifference`. However, _that_ function doesn't appear to be tested: we might also want to write a test for this function.

Reply:
@KtorZ To that end, I've added a small set of tests for `slotDifference` (as far as I can tell, it was previously untested).